File: introducción a POO/ejercicio 3/ej3.java
public class Coche {
    private String marca;
    private String modelo;
    private int velocidad = 0;
    
    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
    }
    
    public void acelerar() {
        velocidad += 10;
    }
    
    public void frenar() {
        velocidad -= 5;
    }
    
    public String toString() {
        return marca + " " + modelo + " - Velocidad: " + velocidad;
    }
    
    public static void main(String[] args) {
        Coche coche1 = new Coche("Toyota", "Corolla");
        Coche coche2 = new Coche("Ferrari", "296GTB");
        
        coche1.acelerar();
        coche1.acelerar();
        coche1.frenar();
        
        coche2.acelerar();
        coche2.frenar();
        coche2.frenar();
        
        System.out.println(coche1);
        System.out.println(coche2);
    }
}

File: introducción a POO/ejercicio 3/ej3.py
class Coche:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = 0
        
    def acelerar(self):
        self.velocidad += 10
    def frenar(self):
        self.velocidad -= 5
        
    def __str__(self):
        return f"{self.marca} {self.modelo} - Velocidad: {self.velocidad}"
        
coche1 = Coche("Toyota", "Corolla")
coche2 = Coche("Ferrari", "296GTB")

coche1.acelerar()
coche1.acelerar()
coche1.frenar()

coche2.acelerar()
coche2.frenar()
coche2.frenar()

print(coche1)
print(coche2)